### ISSUE TITLE
0.3.4 breaks grouping for gulp-inject

### ISSUE THREAD

**Issue by rejas** [doowb/group-array#10]
After updating to v0.3.4 gulp-inject broke. For example:

```[
  {
    file: <File "lib.js">,
    ext: 'js',
    startTag: /"js\b":\s*\[/gi,
    endTag: /\]/gi,
    tagKey: '/"js\\b":\\s*\\[/gi/\\]/gi'
  },
  {
    file: <File "component.html">,
    ext: 'html',
    startTag: /"html\b":\s*\[/gi,
    endTag: /\]/gi,
    tagKey: '/"html\\b":\\s*\\[/gi/\\]/gi'
  },
  {
    file: <File "lib2.js">,
    ext: 'js',
    startTag: /"js\b":\s*\[/gi,
    endTag: /\]/gi,
    tagKey: '/"js\\b":\\s*\\[/gi/\\]/gi'
  },
  {
    file: <File "styles.css">,
    ext: 'css',
    startTag: /"css\b":\s*\[/gi,
    endTag: /\]/gi,
    tagKey: '/"css\\b":\\s*\\[/gi/\\]/gi'
  }
]
```

called with this:

`var filesPerTags = groupArray(files, 'tagKey');`

result nows only in 

```{
  '/js\\b:s*[/gi/]/gi': [
    {
      file: <File "lib2.js">,
      ext: 'js',
      startTag: /"js\b":\s*\[/gi,
      endTag: /\]/gi,
      tagKey: '/"js\\b":\\s*\\[/gi/\\]/gi'
    }
  ],
  '/html\\b:s*[/gi/]/gi': [
    {
      file: <File "component.html">,
      ext: 'html',
      startTag: /"html\b":\s*\[/gi,
      endTag: /\]/gi,
      tagKey: '/"html\\b":\\s*\\[/gi/\\]/gi'
    }
  ],
  '/css\\b:s*[/gi/]/gi': [
    {
      file: <File "styles.css">,
      ext: 'css',
      startTag: /"css\b":\s*\[/gi,
      endTag: /\]/gi,
      tagKey: '/"css\\b":\\s*\\[/gi/\\]/gi'
    }
  ]
}
```

Using v0.3.3 we would get the correct result:

```
{
  '/"js\\b":\\s*\\[/gi/\\]/gi': [
    {
      file: <File "lib.js">,
      ext: 'js',
      startTag: /"js\b":\s*\[/gi,
      endTag: /\]/gi,
      tagKey: '/"js\\b":\\s*\\[/gi/\\]/gi'
    },
    {
      file: <File "lib2.js">,
      ext: 'js',
      startTag: /"js\b":\s*\[/gi,
      endTag: /\]/gi,
      tagKey: '/"js\\b":\\s*\\[/gi/\\]/gi'
    }
  ],
  '/"html\\b":\\s*\\[/gi/\\]/gi': [
    {
      file: <File "component.html">,
      ext: 'html',
      startTag: /"html\b":\s*\[/gi,
      endTag: /\]/gi,
      tagKey: '/"html\\b":\\s*\\[/gi/\\]/gi'
    }
  ],
  '/"css\\b":\\s*\\[/gi/\\]/gi': [
    {
      file: <File "styles.css">,
      ext: 'css',
      startTag: /"css\b":\s*\[/gi,
      endTag: /\]/gi,
      tagKey: '/"css\\b":\\s*\\[/gi/\\]/gi'
    }
  ]
}
```

So only one file is returned when grouping :-( Maybe the updated union-value library has changed?

**Comment by doowb** [comment]
Hi @rejas, thanks for the issue with an example use case. I tracked the issue down to the quotes being removed inside the `tagKey` when using `set-value` (this is used internally inside `union-value` and there was a different update that I didn't notice).

I must not have any tests with quotes like this so I missed the issue here.

If I bump dependencies and publish a `group-array@1.0.0`, would that work for you (`gulp-inject`)? I'll do that, then try to figure out a fix to publish as `group-array@0.3.4`.

**Comment by doowb** [comment]
I published `group-array@1.0.0` which is updated to latest dependencies and I added a test case for this issue.

**Comment by rejas** [comment]
Thx for the fast fix. Works fine now and so I published a new version myself. Cheerio!
